### PR TITLE
Add initial support for psql array streaming

### PR DESCRIPTION
**Pull request by mfussenegger**

### CODE REVIEW

**Comment by mfussenegger** [comment]
@matriv @dobe please review

**Comment by matriv** [comment]
Through

File: sql/src/main/java/io/crate/protocols/postgres/ConnectionContext.java [hunk 1]
```diff
@@ -257,8 +258,9 @@ public void operationComplete(ChannelFuture future) throws Exception {
                     state = State.MSG_HEADER;
                     session = readStartupMessage(buffer);
                     Messages.sendAuthenticationOK(channel);
-                    // TODO: probably need to send more stuff
-                    Messages.sendParameterStatus(channel, "server_version", "95000");
+
+                    // version between 7.3 and 8 to get simpler pg_type queries from jdbc client
```
Comment (dobe) [inline sql/src/main/java/io/crate/protocols/postgres/ConnectionContext.java]: pls add a  note in the documentation about server version value

**Comment by dobe** [comment]
ack

**Pull request merged**